HousingRepairsSchedulingApi.Tests/GatewaysTests/DrsAppointmentGatewayTests.cs

```diff
@@ -618,23 +620,56 @@ public async void GivenAnEndDateEarlierThanTheStartDate_WhenExecute_ThenInvalidE
 
         [Fact]
 #pragma warning disable CA1707
-        public async void GivenValidArguments_WhenExecute_ThenBookingReferenceIsReturned()
+        public async void GivenValidArguments_WhenExecute_ThenOrderIsReturned()
 #pragma warning restore CA1707
         {
             // Arrange
-            const int bookingId = 12345;
+            var bookingReference = "10003829";
+            var startDateTime = new DateTime(2022, 05, 01);
 
             drsServiceMock.Setup(x =>
-                x.CreateOrder(It.IsAny<string>(), It.IsAny<string>(), It.IsAny<string>())
-            ).ReturnsAsync(bookingId);
+                x.SelectOrder(It.IsAny<int>(), It.IsAny<DateTime?>())
+            ).ReturnsAsync(new order { orderId = 863256, theBookings = new booking[] { new booking { bookingId = 12345 } } });
 
             // Act
-            var startDateTime = new DateTime(2022, 05, 01);
-            var actual = await systemUnderTest.BookAppointment(BookingReference, SorCode, LocationId,
+            var actual = await systemUnderTest.BookAppointment(bookingReference, SorCode, LocationId,
                 startDateTime, startDateTime.AddDays(1));
 
             // Assert
-            Assert.Equal(BookingReference, actual);
-        }
+            Assert.Equal(bookingReference, actual);
+        }
+
+        public static IEnumerable<object[]> InvalidOrderTestData()
+        {
+            yield return new object[] { new DrsException(), "10003829", new DateTime(2022, 05, 01), (order)null };
+            yield return new object[] { new DrsException(), "10003829", new DateTime(2022, 05, 01), new order { orderComments = "No bookings in this order" } };
+            yield return new object[] { new DrsException(), "10003829", new DateTime(2022, 05, 01), new order { theBookings = new booking[] { new booking { bookingId = 0 } } } };
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidOrderTestData))]
+#pragma warning disable xUnit1026
+#pragma warning disable CA1707
+        public async void GivenInvalidOrderIsReturned_WhenSelectingAnOrder_ThenExceptionIsThrown<T>(
+            T exception,
+            string bookingReference,
+            DateTime startDateTime,
+            order orderResponse
+            ) where T : Exception
+#pragma warning restore CA1707
+#pragma warning restore xUnit1026
+        {
+            // Arrange
+            drsServiceMock.Setup(x =>
+                x.SelectOrder(It.IsAny<int>(), It.IsAny<DateTime?>())
+            ).ReturnsAsync(orderResponse);
+
+            // Act
+            var act = async () => await systemUnderTest.BookAppointment(bookingReference, SorCode, LocationId,
+                startDateTime, startDateTime.AddDays(1));
+
+            // Assert
+            await act.Should().ThrowExactlyAsync<T>();
```

Comment:
I'm not sure you need to use generics, it would be simpler with the following:
```suggestion
            await act.Should().ThrowAsync<DrsException>();
```

Reply:
Agree with you, corrected